### DISCUSSION TITLE
[Run issue]:The TOPAS extension ProtonRBE is not functioning properly.

### DISCUSSION THREAD

**Discussion by v587aaaaa3** [OpenTOPAS/OpenTOPAS#86] [Run-Problem] [answered]
### Contact Details

1433643725@qq.com

### What was the issue?

What are the reasons that cause ProtonRBE to malfunction, and are there other methods to test RBE and LET?

Welcome to TOPAS, Tool for Particle Simulation (Version 3.9)
Loading parameters starting from: run.txt

Geant4 Data directory has been specified by the
TOPAS_G4_DATA_DIR environment variable as /home/kai/GEANT4/G4DATA

**************************************************************
 Geant4 version Name: geant4-11-01-patch-03 [MT]   (10-November-2023)
  << in Multi-threaded mode >> 
                       Copyright : Geant4 Collaboration
                      References : NIM A 506 (2003), 250-303
                                 : IEEE-TNS 53 (2006), 270-278
                                 : NIM A 835 (2016), 186-225
                             WWW : http://geant4.org/
**************************************************************

TOPAS is in MT mode, setting number of threads to: 8

TOPAS set the tolerances based on the World size to:
      Surface tolerance = 0.022 nm, and radial tolerance = 0.022 nm.
Checking overlaps for volume Phantom:0 (G4Box) ... OK! 
Checking overlaps for volume Sample:0 (G4Box) ... OK! 

Checking for Unused Components.
These are Components that were never built even though they were at least
partially defined in some parameters. This is not necessarily an error.
It may be an intentional choice to have these Component definitions handy for later use.
To turn off this check, set Ge/CheckForUnusedComponents to False
  No unused components were found.

/home/kai/GEANT4/G4DATA/G4NDL4.7
@@@ G4ParticleHPInelastic instantiated for particle neutron data directory variable is G4NEUTRONHPDATA pointing to /home/kai/GEANT4/G4DATA/G4NDL4.7/Inelastic
@@@ G4ParticleHPInelasticData instantiated for particle neutron data directory variable is G4NEUTRONHPDATA pointing to /home/kai/GEANT4/G4DATA/G4NDL4.7
NeutronHP: /Capture file for Z = 6, A = 12 is not found and NeutronHP will use /home/kai/GEANT4/G4DATA/G4NDL4.7/Capture/CrossSection/6_nat_Carbon
NeutronHP: /Elastic file for Z = 6, A = 12 is not found and NeutronHP will use /home/kai/GEANT4/G4DATA/G4NDL4.7/Elastic/CrossSection/6_nat_Carbon
NeutronHP: /Inelastic file for Z = 6, A = 12 is not found and NeutronHP will use /home/kai/GEANT4/G4DATA/G4NDL4.7/Inelastic/CrossSection/6_nat_Carbon
Topas is exiting due to a serious error in scoring setup.
Sc/Carabe/Quantity has unknown value: RBE_Carabe

### TOPAS category

Scoring

### Version

3.9

### What Operative System (OS) are you running on?

Linux

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by jschuemann** [comment]
Hi,
you are running with an outdated version of TOPAS, please use version OpenTOPAS v4.0 (see installation guides on this website), and it looks like you may have not compiled TOPAS with the RBE extensions, so TOPAS does not know about the RBE model. Please follow the instructions on adding extensions in our documentation to v4.0.

**Reply by v587aaaaa3** [reply]
I'm sorry to bother you again. I have now updated the TOPAS version to 4.0. I noticed that clicking on the GitHub link on this website (https://opentopas.readthedocs.io/en/latest/extensions/rbe/intro.html) does not redirect to the corresponding URL for ProtonRBE.

Additionally, I would like to ask how to compile ProtonRBE into TOPAS. I have successfully compiled the Microdosimetry Extension into TOPAS; do I need to use a similar method to compile ProtonRBE into TOPAS. Does TOPAS support RBE and LET for other particles? I have found that currently TOPAS seems to only support RBE and LET for protons.

**Reply by jschuemann** [reply]
Yes, every time you add an extension, you need to recompile TOPAS for that extension.

**Reply by v587aaaaa3** [reply]
Hello, I would like to know if TOPAS provides information about the scoring for RBE and LET for other particles, such as carbon ions？

**Comment by jschuemann** [comment, marked as answer]
There are codes by others that do that, our default does not yet include LET for other particles. Some of the RBE scorers can use other particles (see RBE documentation, mostly scorers that don't use LET). Carbon LET and other properties will be added at some point, or you can also develop your own scorer, e.g., based on the proton LET scorer.
